src/irmin-pack/unix/utils.ml

```diff
@@ -63,22 +63,24 @@ end = struct
     (!(t.nb_contents), !(t.nb_nodes), !(t.nb_commits))
 end
 
-(** [nearest_leq ~arr ~get ~lo ~hi ~key] returns the nearest entry in the sorted
-    [arr] that is [<=] the given key. Routine is based on binary search. *)
-let nearest_leq ~arr ~get ~lo ~hi ~key =
-  assert (lo <= hi);
-  if get arr lo > key then `All_gt_key
-  else if get arr hi <= key then `Some hi
-  else
-    let interval lo hi =
-      assert (get arr lo <= key && key < get arr hi);
-      assert (lo < hi);
-      (lo, hi)
-    in
-    interval lo hi
-    |> iter_k (fun ~k (lo, hi) ->
-           if lo + 1 = hi then `Some lo
-           else
-             let mid = (lo + hi) / 2 in
-             if get arr mid <= key then k (interval mid hi)
-             else k (interval lo mid))
+(** [nearest_geq ~arr ~get ~lo ~hi ~key] returns the index of the nearest entry
+    in the sorted [arr] that is [>=] the given key. Routine is based on binary
+    search. *)
+let nearest_geq ~arr ~get ~lo ~hi ~key =
```

Comment:
Why did you change from `leq` to `geq`?

Reply:
I needed to support traversing the sparse file from left to right (for `Traverse_pack_file` and `irmin-tezos-utils`), jumping from one "object" `(off,len)` to the next `(off+len,...)` but skipping over the sparse holes... so I first added `geq` then realized that `leq` was redundant :) (`geq` gives you the next valid offset, while `leq` would have worked for a right-to-left traversal)

Reply:
Ah, okay. That history makes sense!